package.json

```diff
@@ -1,13 +1,82 @@
 {
-  "name": "vibc-core-smart-contracts",
-  "version": "1.0.0",
-  "main": "index.js",
-  "repository": "https://github.com/open-ibc/vibc-core-smart-contracts",
+  "name": "@open-ibc/vibc-core-smart-contracts",
+  "version": "0.1.3",
+  "main": "dist/index.js",
+  "bin": {
+    "deploy-vibc-core-smart-contracts": "./dist/scripts/deploy-script.js",
+    "upgrade-vibc-core-smart-contracts": "./dist/scripts/upgrade-script.js"
+  },
   "license": "MIT",
   "dependencies": {
-    "solhint": "^4.1.1"
+    "@commander-js/extra-typings": "^12.0.1",
+    "@typechain/ethers-v6": "^0.5.0",
+    "ethers": "^6.7.1",
+    "nunjucks": "^3.2.4",
+    "solhint": "^4.1.1",
+    "typechain": "^8.3.2",
+    "winston": "^3.13.0",
+    "yaml": "^2.4.1",
+    "zod": "^3.23.4",
+    "zx": "^8.0.2"
+  },
+  "devDependencies": {
+    "@nomicfoundation/hardhat-chai-matchers": "^2.0.0",
+    "@nomicfoundation/hardhat-ethers": "^3.0.0",
+    "@nomicfoundation/hardhat-ignition": "^0.15.0",
+    "@nomicfoundation/hardhat-ignition-ethers": "^0.15.0",
+    "@nomicfoundation/hardhat-network-helpers": "^1.0.0",
+    "@nomicfoundation/hardhat-toolbox": "^5.0.0",
+    "@nomicfoundation/hardhat-verify": "^2.0.0",
+    "@typechain/hardhat": "^9.0.0",
+    "@types/nunjucks": "^3.2.6",
+    "@types/winston": "^2.4.4",
+    "chai": "^4.2.0",
+    "hardhat": "^2.14.0",
+    "hardhat-gas-reporter": "^1.0.8",
+    "solidity-coverage": "^0.8.0",
+    "tsup": "^8.0.2"
   },
   "scripts": {
-    "lint": "solhint contracts/**/*.sol"
+    "lint": "solhint contracts/**/*.sol",
+    "test": "forge test",
+    "build": "npm run gen-types && tsup",
+    "gen-types": "npm run build-contracts && typechain --target ethers-v6 --out-dir src/evm/contracts/ './out/?(OptimisticProofVerifier|ProofVerifier|Ibc|IbcUtils|Channel|Dispatcher|Mars|Earth|UniversalChannelHandler|DummyProofVerifier|DummyLightClient|ERC1967Proxy|OptimisticLightClient).sol/*.json'",
+    "build-contracts": "forge build",
+    "deploy-contracts": "npm run build && node dist/deploy.js",
+    "deploy-simple": "node dist/deploy.js",
+    "prepublish": "npm run build"
```

Comment:
can we also make sure we build the bindings before we publish? Maybe this is not the right place for that but it would be nice to add some checks that will prevent us from "forgetting about updating the bindings" with each release.

Reply:
not sure I understand your comment, `npm run build` calls `npm run gen-types && tsup`, and `npm run gen-types` generates the bindings. Since we've included `npm run build` in our prepublish, it should always run this command before publishing? So I don't think we need to check for updating bindings, since they should automatically be done on publish

Reply:
I didn't realise that `npm run build` called the binding construction target as well. Maybe it should be renamed to `build-contracts-and-bindings` ?

Reply:
npm run build also runs `tsup`, which also bundles _all_ typescript files into javascript in a releasable version (which is also another step we need to do before publishing). So since it also bundles the whole package, I don't think calling it `build-contracts-and-bindings` is accurate? I think it's convention with npm packages to have the generic "build" command build _everything_, including the typescript files _and_ any bindings/types/modules etc

 We already do have a `gen-types` command which might be what you are thinking of, I think we can rename that to `build-contracts-and-bindings` in case that would be clearer naming

Reply:
renamed the gen-types to `build-ts-contract-bindings`! Though let me know if this can still be made clearer!